auth/auth.go

```diff
@@ -227,6 +234,9 @@ type CredentialsOptions struct {
 	QuotaProjectIDProvider CredentialsPropertyProvider
 	// UniverseDomainProvider resolves the universe domain with the credentials.
 	UniverseDomainProvider CredentialsPropertyProvider
+	// IsServiceAccount indicates if the token provider uses a Google Cloud service account.
+	// When true, service account specific trust boundary logic is enabled.
+	IsServiceAccount bool
```

Comment:
I'm wondering if it would be possible to replace this field with a new `TrustBoundaryProvider` or `TrustBoundaryOptions` type and field at this location. The goal would be to replace the 3 new fields in `Options2LO` and the 2 new fields in `impersonate.Options` with a single type that holds everything needed to execute `LookupServiceAccountTrustBoundary` and cache the result. This may not turn out to be feasible. The thought here is simply to aggregate the configuration for  `LookupServiceAccountTrustBoundary` as early as possible, keeping everything associated with this operation bundled together.

Reply:
Implemented a trust boundary provider.
Let me know what you think!